nucypher/policy/policies.py

```diff
@@ -161,6 +169,130 @@ def payload(self):
         return bytes(self.publish_transaction) + partial_payload
 
 
+class NodeEngagementMutex:
+    """
+    TODO: Does this belong on middleware?
+
+    TODO: There are a couple of ways this can break.  If one fo the jobs hangs, the whole thing will hang.  Also,
+       if there are fewer successfully completed than percent_to_complete_before_release, the partial queue will never
+       release.
+
+    TODO: Make registry per... I guess Policy?  It's weird to be able to accidentally enact again.
+    """
+    log = Logger("Policy")
+
+    def __init__(self,
+                 f,    # TODO: typing.Protocol
```

Comment:
More descriptive name...? I give `self.f` an `F` 😂 . 

Maybe `node_f` or `node_func` or `node_action`.

Reply:
Well, `f` is a traditional name for a function passed to a wrapper with intent to call it.  But yeah, I can change it to... something better.